python/ml4ir/base/features/feature_config.py

```diff
@@ -116,32 +116,27 @@ def extract_features(self):
         - label
         - all_features
         """
-        try:
-            self.query_key = self.features_dict.get(FeatureConfigKey.QUERY_KEY)
-            self.all_features.append(self.query_key)
-        except KeyError:
```

Comment:
I think KeyError would never happen with this code. features_dict[foo] would raise KeyError, but features_dict.get(foo) just return None.

So I'm now checking upfront and explicitly raising an error.

Reply:
you are right. `dict.get` returns `None` be default. Good catch!